### ISSUE TITLE
Input seriously lagging in UWP

### ISSUE THREAD

**Issue by MartinZikmund** [MonoGame/MonoGame#4897]
Keyboard and mouse input is seriously lagging in UWP games.
When you use `Keyboard.GetState().IsKeyDown( ... )` you will find out that it returns true usually even several Updates after the key is released.
Similar lag occurs for mouse state.

This is quite serious as it renders the framework unusable - or rather the games hardly playable.

Edit: I have confirmed this behavior with another developer, so it is not just an issue on my side.

**Comment by Ragath** [comment]
I believe I have encountered this issue too, although in my case the keys appear to be permanently stuck until you press another key or move the mouse.

**Comment by VitezslavImrysek** [comment]
Same problem here. This issue affects `TouchPanel` as well.
My guess is that the issue is somehow connected to how MonoGame interacts with `Dispatcher` (and therefore input event queue processing) as it also affects `CoreWindow` `KeyUp`/`KeyDown` events and also `GetKeyState` method.

**Comment by tomspilman** [comment]
I'll give this a look on my laptop later tonight.

**Comment by tomspilman** [comment]
So I made a little test app gathering keyboard state in Update and drawing the string in Draw.  I saw no input lag of any kind.

Note i'm running Windows 10 Pro with version 10.0.10586.318 and the 

@MartinZikmund @VitezslavImrysek  - What version of Windows 10 are you on?

**Comment by tomspilman** [comment]
Here is my test code:

Update:
```
            var kstate = Keyboard.GetState();

            downKeys = string.Empty;
            var keys = kstate.GetPressedKeys();
            foreach (var k in keys)
                downKeys += k + ",";

            if (kstate.IsKeyDown(Keys.OemQuestion))
                downKeys += "?";
```

Draw:
```
           spriteBatch.Begin(sortMode:SpriteSortMode.Immediate);
           spriteBatch.DrawString(font, "Keys: " + downKeys, new Vector2(200, 300), Color.White);
           spriteBatch.End();
```

**Comment by Danthekilla** [comment]
We are having the same issue I think.

Here is a video of the issue with my input visible from a input displaying program.

When I press Keys.Up (via Keyboard.GetState(); which eventually hooks into CoreWindow.KeyDown) I move the background forward and back. You can also see the input system moving the selection caret.

https://www.youtube.com/watch?v=vHr91Sr_W_g&feature=youtu.be

As you can see when I move the mouse it delays the KeyUp (and KeyDown) events from triggering. I have tried 3 different ways of getting mouse movement and they all affect it the same. When I don't move the mouse there is no lag in keyboard input most (95%) of the time.

What do you make of this @tomspilman ?

I could try and get a test case working if that would be useful.

I am on the latest build of windows 10 (insider preview fast ring) and we have also tested it on a machine with the current release branch.

**Comment by VitezslavImrysek** [comment]
Yes, this is the same kind of issue me and Martin were experiencing. 
From what I can tell this issue started to show up only after some game logic is put in - Some stuff to render, some stuff to do in Update. Thats why it didnt show up in tomspilman's test case - that test case was just too "simple". At least thats what I can tell based on my observations.
Again, this was definately a bug at the time of reporting as the very same code was running just fine in Win32 environment.

**Comment by tomspilman** [comment]
Fixed in #5520.
